webpack/assets/javascripts/react_app/components/users/PersonalAccessTokens/PersonalAccessTokenForm.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import * as Yup from 'yup';
+import { Button } from 'patternfly-react';
+import ForemanForm from '../../common/forms/ForemanForm';
+import TextField from '../../common/forms/TextField';
+import ForemanModal from '../../ForemanModal';
+import { useForemanModal } from '../../ForemanModal/ForemanModalHooks';
+import { maxLengthMsg, requiredMsg } from '../../common/forms/validators';
+import { translate as __ } from '../../../common/I18n';
+import { submitForm } from '../../../redux/actions/common/forms';
+
+const MODAL_ID = 'personal-access-tokens-form-modal';
+
+const tokenFormSchema = Yup.object().shape({
+  name: Yup.string()
+    .max(...maxLengthMsg(254))
+    .required(requiredMsg()),
+  expires_at: Yup.date().min(new Date(), __('Cannot be in the past')),
+});
+
+const PersonalAccessTokenForm = ({ controller, url, initialValues }) => {
+  const dispatch = useDispatch();
+  const { setModalOpen, setModalClosed } = useForemanModal({
+    id: MODAL_ID,
+  });
+
+  const handleSubmit = async (values, actions) => {
+    await dispatch(
+      submitForm({
+        url,
+        values: { ...values, controller },
+        item: 'personal_access_token',
+        message: __('Personal Access Token was successfully created.'),
+      })
+    );
+    setModalClosed();
+  };
+
+  return (
+    <p>
+      <Button bsStyle="success" onClick={setModalOpen}>
+        {__('Add Personal Access Token')}
+      </Button>
+
+      <ForemanModal id={MODAL_ID} title={__('Create Personal Access Token')}>
+        <ForemanModal.Header />
+        <ForemanForm
+          onSubmit={(values, actions) => handleSubmit(values, actions)}
+          initialValues={initialValues}
+          validationSchema={tokenFormSchema}
+          onCancel={setModalClosed}
+        >
+          <TextField
+            name="name"
+            type="text"
+            label={__('Name')}
+            inputClassName="col-md-6"
+            required
+          />
+          <TextField
+            name="expires_at"
+            type="datetime-local"
+            required={false}
+            label={__('Expires')}
+            inputClassName="col-md-6"
+          />
```

Comment:
Oh, I did not noticed this. Why we are not using our dateTime picker?
This won't work on browsers that doesn't support `datetime-local` field.
We can use our component from `'../../common/form/DateTimePicker'`
```suggestion
          <DateTimePicker
            name="expires_at"
            required={false}
            label={__('Expires')}
            inputClassName="col-md-6"
          />
```